secretsmanager/src/misc.h

```diff
@@ -0,0 +1,69 @@
+// This implementation is copied from CryptoPP: https://www.cryptopp.com/wiki/OPENSSL_EVP_BytesToKey
```

Comment:
Why are we copying this?

Reply:
By the initial design we use a password based key derivation algorithm from openssl which is now deprecated. I'm discussing this with Ittai and Sanchita and I hope we will remove this.

About your question - the Crypto++ folks say "This algo is depricated, so we have got no implementation. If you insist on using it - here is a header which you can put in your source tree." 

Even this warning sounds scary enough to me so we change the key derivation algorithm.

Reply:
Also we won't need this header once we remove the key generation step.